cpu/esp32/Makefile.include

```diff
@@ -86,9 +59,19 @@ LINKFLAGS += -T$(RIOTCPU)/$(CPU)/ld/esp32.rom.ld
 LINKFLAGS += -T$(RIOTCPU)/$(CPU)/ld/esp32.rom.nanofmt.ld
 LINKFLAGS += -nostdlib -lgcc -Wl,-gc-sections
 
-ifneq (,$(filter esp_cxx,$(USEMODULE)))
-  UNDEF += $(BINDIR)/esp_cxx/cxa_guard.o
-endif
+# Baselibs needed when using esp_wifi_any pseudomodule
+_ESP_WIFI_ANY_BASELIBS = -lcore -lrtc -lnet80211 -lpp -lcoexist -lwps -lwpa -lwpa2
+_ESP_WIFI_ANY_BASELIBS += -lphy -lstdc++
+
+# Baselibs needed when using esp_now module
+_ESP_NOW_BASELIBS = $(_ESP_WIFI_ANY_BASELIBS) -lespnow -lmesh
+
+# NOTE: These checks can be turned into a normal conditional when #9913 is fixed
+BASELIBS += $(if $(filter esp_wifi,$(USEMODULE)),$(_ESP_WIFI_ANY_BASELIBS))
+BASELIBS += $(if $(filter esp_now,$(USEMODULE)),$(_ESP_NOW_BASELIBS))
```

Comment:
```suggestion
# Baselibs needed when using esp_now module
_ESP_NOW_BASELIBS = -lespnow -lmesh

# NOTE: These checks can be turned into a normal conditional when #9913 is fixed
BASELIBS += $(if $(filter esp_wifi_any,$(USEMODULE)),$(_ESP_WIFI_ANY_BASELIBS))
BASELIBS += $(if $(filter esp_now,$(USEMODULE)),$(_ESP_NOW_BASELIBS))
```
In this case `esp_wifi_any` would pull in the base libraries, also if `esp_now` is enabled.

Reply:
Changed